keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,682 @@
+
+#### Config File Permissions
+
+  *Conformance Profile: Essential*
+
+  The configuration files for API server, controller manager, scheduler and kubelet should have the correct setup of ownership and file permissions.
+  Without proper permission setup, a local user will be able to access prohibited information, including
+  - private keys
+  - secrets
+  - credentials
+
+  The criteria are:
+  - etcd data directory is owned and accessible only by the user running etcd;
+  - kubeconfig files for control plane is owned and accessible only by the user runs the control plane;
+  - private key files is owned and accessible only by the user who runs the process which consumes the key;
+  - kubeconfig for administration purpose (if present, e.g. `/etc/kubernetes/admin.conf`)
+    is owned and accessible only by the admin user (perm 0400)/group (perm 0440)
+    (the test doesn't verify particular user/group, but only the permissions).
+
+
+  Notes:
+  - the above statements exclude _root_ user who always have full access to the system.
+  - if etcd is managed separately, the test may not cover it.
+
+
+##### Test methodology
+
+  These tests will require the accessibility to the host.
+  There are a few options to achieve that:
+
+    - Create a Pod with special privilege (e.g. in `kube-system` namespace) and
+    run with host pid namespaces, bind mount host file system into testing container for inspection.
+    E.g. creating a Pod using DaemonSet.
+    This is the most consistent way with other tests without requiring user to perform additional operations before running the tests.
+
+    - Remote access to master, like SSH.
+    This is the alternative way if it's impossible to schedule a Pod on master (the kubelet on master doesn't register).
+
+  The test code will inspect the master/node system by first looking up the processes by names.
+  The names are default to `etcd`, `kube-apiserver`, `kube-controller-manager`, `kube-scheduler`, `kubelet`.
+  In the case when the cluster admin changes these names,
+  overriding mechanism (e.g. via environment variable) must be supported by test code.
+
+  With the processes identified by the names,
+  test code will be able to validate command line flags by looking into `/proc` file system (must run in host pid namespace), and also find out the user running these processes.
+  From the command line flags, the test code will identify the config/cert/key files being used,
+  and validate the file permission from there.
+
+  This test or part of the test can be opt-out in the case host access is impossible.
+
+#### API privileges of default service accounts
+
+  *Conformance Profile: Essential*
+
+  Every namespace has a service account named `default`.
+  Without explicit permission grant from RBAC RoleBinding or ClusterRoleBinding,
+  This service account has the minimum privilege to access Kubernetes API.
+  The cluster is vulnerable if a malicious Pod can be scheduled with default service account
```

Comment:
nit: "The cluster is vulnerable..." depends on the threat model. I'd just leave this out.